### DISCUSSION TITLE
Gastronomy dataset is deprecated

### DISCUSSION THREAD

**Discussion by d3liaa** [noi-techpark/it.bz.opendatahub.databrowser#307] [Q&A]
The gastronomy dataset is deprecated, and the same info is in activity and POIs. So should we remove it from the databrowser?

**Comment by gappc** [comment]
@d3liaa you are right and its very good that you bring up this topic.

The more general question is: what should we do with deprecated endpoints (e.g. should we show them)?

@sseppi @RudiThoeni what is your opinion?

**Reply by sseppi** [reply]
Are we able to understand if a dataset is deprecated from the API?

If yes, I would keep the Dataset in the Data Browser and mark it as deprecated.

**Comment by gappc** [comment]
@sseppi yes we get the deprecation information from the MetaData API and from OpenAPI (Swagger) too. How should we visualize that a dataset is deprecated?

@RudiThoeni do we have to take special care of deprecated datasets? Is there something extra to be known?

**Comment by sseppi** [comment]
@gappc I would suggest to add a label near the "Dataset name" or something that indicates that is deprecated and then add a tooltip that explain what it means. But here maybe @pkritzinger has a better idea on how to do.

I would keep the OpenAPI deprecation information so we have less effort in maintaining it. @RudiThoeni do you agree?

**Comment by gappc** [comment]
At the moment I show the text "(Deprecated)" near the title in dataset-overview. At the moment, the deprecation information is not shown anywhere else (like in Table-, Detail-, Edit-, RawView).

@sseppi if I understand you right, you suggest to use something like the labels / tags as shown in the image below? If so, should we append that label / tag to the title? @pkritzinger would that be a good solution?

![image](https://user-images.githubusercontent.com/5732768/221147723-497455b6-c27e-40f1-b045-b846681dd600.png)

Regarding the other topic:

> I would keep the OpenAPI deprecation information so we have less effort in maintaining it. @RudiThoeni do you agree?

I'm not sure if it is a good idea to mix MetaData API and OpenAPI information only to get the deprecation information, because then a lot of edge cases emerge that need proper handling. In addition, through OpenAPI it is only possible to deprecate a whole endpoint, but it would not be possible anymore to deprecate datasets like "NOI Article".

If there is the general requirement to mix MetaData API and OpenAPI data, then we have to discuss that in detail.

@sseppi, @RudiThoeni: please let me know what you think about this.

**Reply by sseppi** [reply]
> At the moment I show the text "(Deprecated)" near the title in dataset-overview. At the moment, the deprecation information is not shown anywhere else (like in Table-, Detail-, Edit-, RawView).
> 
> @sseppi if I understand you right, you suggest to use something like the labels / tags as shown in the image below? If so, should we append that label / tag to the title? @pkritzinger would that be a good solution?
> 
> ![image](https://user-images.githubusercontent.com/5732768/221147723-497455b6-c27e-40f1-b045-b846681dd600.png)
I think so.
> 
> Regarding the other topic:
> 
> > I would keep the OpenAPI deprecation information so we have less effort in maintaining it. @RudiThoeni do you agree?
> 
> I'm not sure if it is a good idea to mix MetaData API and OpenAPI information only to get the deprecation information, because then a lot of edge cases emerge that need proper handling. In addition, through OpenAPI it is only possible to deprecate a whole endpoint, but it would not be possible anymore to deprecate datasets like "NOI Article".
> 
> If there is the general requirement to mix MetaData API and OpenAPI data, then we have to discuss that in detail.
> 
> @sseppi, @RudiThoeni: please let me know what you think about this.

In my point of view the deprecation is only regarding Endpoints that are old and will be shut down in a short future (e.g. the v1 of the API will be deprecated when a v2 comes out) and all Dataset that has been created as child of the endpoint should be deprecated.
For this reason I suggested to use the information that comes from the OpenAPI. But let hear what the opinion of Rudi is.

**Comment by pkritzinger** [comment]
hi @gappc @sseppi we need to take a closer look on how to visualize the deprecation info. 

Administrative remark. From my point of view, there are two issues/ discussions going on here:
1. How to visualize deprecation info.
2. Where to integrate deprecation info.

Can we make two issues out of them and plan them within one of the releases?

**Comment by RudiThoeni** [comment]
Hi everybody

I would go fo the simple solution, do not show deprecated Datasets in the databrowser.....
I want to geat rid of this endpoints since years.... and its not easy to remove Endpoints used by customers,.... the more we display it more it will be used.... so my point is we do not show it only in swagger....

Let's keep the openapi deprecated info, and maybe for the metadata api we simply remove it?

**Reply by sseppi** [reply]
Hi Rudi,
thank you for the answer.

> Hi everybody
> 
> I would go for the simple solution, do not show deprecated Datasets in the databrowser..... I want to geat rid of this endpoints since years.... and its not easy to remove Endpoints used by customers,.... the more we display it more it will be used.... so my point is we do not show it only in swagger....

I got your point, and for me is fine to not show the deprecated endpoint at all in the Data Browser. If we do so we have to be worry about the Dataset that are in the Metadata API related to a Deprecated Dataset... In my point of view they should be deprecated as wee and follow the same logic of the Enpoint.

@ohnewein  would like to see in the Data Browser all data exposed at an  API level. So, to take a final decision, we need to talk with him and decide what to do considering also his feedback.
> 
> Let's keep the openapi deprecated info, and maybe for the metadata api we simply remove it?

Fine for me, since in the Metadata API we have the relationship between Endpoint and Dataset, and so the Dataset could inherit the configuration of the parent Endpoint.

**Reply by gappc** [reply]
@sseppi @RudiThoeni regarding the MetaData API: we are free to ignore entries in MetaData API where the deprecation flag is set, so if you want you can leave the deprecated datasets / endpoints in MetaData API and we filter them out, or you remove them from MetaData API alltogether. In the second case I suggest to remove the `Deprecated` field from the MetaData API as it is not needed.

Please let me know about your decision.
